app/utils/push_proxy.ts

```diff
@@ -23,7 +37,14 @@ export function canReceiveNotifications(serverUrl: string, verification: string,
     }
 }
 
-export function alertPushProxyError(intl: IntlShape) {
+const handleAlertResponse = async (buttonIndex: number, serverUrl: string) => {
+    if (buttonIndex === 0) {
```

Comment:
There is only one button, and always sets the ack, so... how would it show the warnings in the rest of the product? Shouldn't be two buttons like "OK" and "Dont remember me anymore"? And maybe that second button only available when checking the warning after login? 0/5

Reply:
the way I approached is like, for users that are already logged in the warning will still show once logged in and they will be able to discard it once clicked "ok". For those who log in for the first time the warning will show just once and will reduce the annoying part of showing it in different places and several times.

Reply:
@esethna Thoughts about this?

Reply:
@esethna so, there are basically three places where we inform user that the push can not be received due to configuration: initial alert window on loading the app, channels header alert icon, servers list alert icon.

Current implementation works like this:
- first app loading shows users the alert window notifying that due to configuration the push alerts can not be received. Users clicks ok, log in and then the header bar and the server show the icon which if clicked will show the Alert again, and again, etc.

My approach with the changes in this PR is:
- For existing users that are already logged in in a server which has the push disabled, the alert icon in the header and server list will be shown, once they click it and the alert window is shown and click 'ok' acknowledging the warning, these alert icons won't show again.
- For new users the initial alert window will be shown, but once discarded (which is forcefully discarded since there is only one button) the other two places (channel header and server list) won't ever be shown since the alert was already acknowledged.

So, what @larkox is proposing is that this Alert Window should have two buttons, "Ok" meaning like just hiding the window and the "Don't remember again" which will hide the alert icons from the channels header and server lists and won't show the Alert again. We would like to read your thoughts on this. Thanks!

Reply:
@johndavidlugtu looping you in here for this conversation given it's in the realm of potentially missing notifications due to server configuration (ie push notifications are disabled) and this PR makes some changes to how we are informing end users about that.

Reply:
Thanks @esethna 

@pvev I'm +1 on only showing the the alert window on login only once and having it dismissed once "Okay" is clicked. That said, I'm not sure dismissing the alert icon for good after clicking "Okay" once is ideal.
<img width="52" alt="Screenshot 2023-09-01 at 11 01 31 AM" src="https://github.com/mattermost/mattermost-mobile/assets/101303489/2afcf62d-5cee-4bb8-806d-e7b0970bfb99">

To @larkox 's point, we want to preserve the ability to keep the warning somewhere in the product. As push notifications is a pretty integral part of the mobile app experience, I'd argue keeping the alert icon persistent is a good way to note to users that they do not have the complete mobile app experience. Clicking on that icon should pop up the alert window. In this case, the alert window isn't annoying since the info is revealed on-demand.

Thoughts?

Reply:
The reason to dismiss it is so it is not so distracting. Having a red warning in the screen causes certain anxiety for some people like "this has to be fixed" (similar with having unread mentions).

Also, something that is not shown in the videos, is that (if I remember correctly) the server list has a lengthy text below the server stating the problem that server has. @pvev could you verify that and show how that text looks here for John to see? You may have to scroll a bit in the server list, or add another server, to properly see it (the bottom sheet should take those errors when creating the initial size, but it is not right now).

Reply:
Since push notifications is a big component of the value of the mobile app, I do think there's value in having users feel that there's something that "has to be fixed" and can tell their admins to set it up.

That said, I'm curious -- @esethna have you heard from users complaining about this icon specifically?

Re: the second paragraph, will wait for @pvev to comment. I believe he's on PTO at the moment

Reply:
so @johndavidlugtu @esethna following is the text that is shown when you open the server list and the alert is still shown to users:

<img width="431" alt="Screenshot 2023-10-02 at 13 58 38" src="https://github.com/mattermost/mattermost-mobile/assets/10082627/2ba156a2-db17-4ae8-a011-8b1ec22f4baa">

Reply:
so, @johndavidlugtu @esethna,  based on the conversation thread, should we go then with:

1) on server login, show the notification and once acknowledged, don't show it again.

2) inside the app, keep the alert icon (which will trigger the alert if touched) so the users know that there is something going on with their configuration that needs to be attended. * clicking ack in the alert shown once inside the app won't hide the alert icon.

Is that the way we want this to work?

Reply:
That's correct, yes. We want to preserve the ability for end users to have a passive alert icon letting them know of the configuration issue -- similar to how a car having a "check engine" light until resolved.

This also makes it so users are aware the reason they're not receiving push notifications isn't because of a bug

Reply:
ok, understood, will proceed with that change then and update this PR. thanks @johndavidlugtu 👍

Reply:
hey @esethna @johndavidlugtu the change is now ready and working as expected:


https://github.com/mattermost/mattermost-mobile/assets/10082627/1c62a9ec-dfcf-4c95-974c-2e71daa6703b

Reply:
hey @johndavidlugtu @esethna friendly reminder on this one :)

Reply:
Thanks for the ping @pvev 

Can we update the copy to say "Due to the configuration of this server..."
<img width="285" alt="Screenshot 2023-10-17 at 7 07 52 PM" src="https://github.com/mattermost/mattermost-mobile/assets/101303489/02451239-eb41-49b5-8da6-dedc2c1c0e73">

Otherwise, lgtm!

Reply:
done @johndavidlugtu 
<img width="357" alt="Screenshot 2023-10-18 at 20 57 58" src="https://github.com/mattermost/mattermost-mobile/assets/10082627/96dab770-96cc-4376-b667-203c1a60b812">